### PR TITLE
Make download_action command write a script to execute the action loc…

### PR DESCRIPTION
**Pull request by gkousik**
…ally

This will be useful if the developer just wants to run the action within the docker container on their machine after downloading the action inputs

### CODE REVIEW

File: go/pkg/tool/tool.go [hunk 1]
```diff
@@ -393,6 +398,55 @@ func (c *Client) DownloadAction(ctx context.Context, actionDigest, outputPath st
 	return c.writeProto(is, filepath.Join(outputPath, "input_node_properties.textproto"))
 }
 
+func (c *Client) writeExecScript(ctx context.Context, cmd *repb.Command, filename string) error {
+	if cmd == nil {
+		return fmt.Errorf("invalid comment (nil)")
```
Comment (mrahs) [inline go/pkg/tool/tool.go:403]:
```suggestion
		return fmt.Errorf("invalid command (nil)")
```

File: go/pkg/tool/tool.go [hunk 2]
```diff
@@ -393,6 +398,55 @@ func (c *Client) DownloadAction(ctx context.Context, actionDigest, outputPath st
...
+	}
+
+	var runActionScript bytes.Buffer
+	runActionFilename := filepath.Join(filepath.Dir(filename), "run_command.sh")
+	wd := cmd.WorkingDirectory
+	execCmd := strings.Join(cmd.GetArguments(), " ")
+	runActionScript.WriteString(fmt.Sprintf("#!/bin/bash\n\n"))
```
Comment (mrahs) [inline go/pkg/tool/tool.go:410]: Unnecessary `fmt`?

Maybe `fmt.Fprintf(runActionScript, "#!/bin/bash\n\n")`?

File: go/pkg/tool/tool.go [hunk 3]
```diff
@@ -393,6 +398,55 @@ func (c *Client) DownloadAction(ctx context.Context, actionDigest, outputPath st
...
+	runActionScript.WriteString(fmt.Sprintf("# This script is meant to be called by %v.\n", filename))
```
Comment (mrahs) [inline go/pkg/tool/tool.go:411]: "by" implies an actor. "from within" is probably clearer?

File: go/pkg/tool/tool.go (hunk 1 above)
Comment (gkousik) [reply go/pkg/tool/tool.go:403]: Done.

File: go/pkg/tool/tool.go (hunk 2 above)
Comment (gkousik) [reply go/pkg/tool/tool.go:410]: Done.

File: go/pkg/tool/tool.go (hunk 3 above)
Comment (gkousik) [reply go/pkg/tool/tool.go:411]: Done.

**Review by mrahs — APPROVED** [review]
It looks like there are some hashes in the test that need to be updated now that there is an extra file in the output.

**Pull request merged**